### PR TITLE
Refactor and simplify KeyManager API

### PR DESCRIPTION
**Pull request by msfjarvis**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [x] Refactoring
- [ ] Dependency updates

## :scroll: Description

Simplifies the `KeyManager` API surface to operate on a singular `Key` type, which elides the public/private distinction to be handled solely by `KeyManager` subclasses.

Refactors `PGPKeyManager` to adapt to this new API, taking the opportunity to land some important fixes

- Key IDs are now correctly hex-encoded
- `PGPKeyManager#getKeyById` now handles both user/email and key ID as a valid ID to find keys using.

## :bulb: Motivation and Context

Wrapping a key into a class like `KeyPair` created a litany of problems that really made things impossible to work with

- Forcing every instance of `KeyPair` to have a 'secret' part meant you could not add encrypt-only keys
- By making 'key id' a part of `KeyPair` meant you could not extract things like user IDs and key fingerprints which are necessary for PGP-based key finding

## :green_heart: How did you test it?

Unit tests still pass

## :pencil: Checklist

- [x] I formatted the code `./gradlew spotlessApply`
- [x] I reviewed submitted code
- [ ] I added a [CHANGELOG](CHANGELOG.md) entry if applicable

## :crystal_ball: Next steps

Rebase #1647 and decide on the next logical unit to separate out of that PR.

## :camera_flash: Screenshots / GIFs

N/A

### CODE REVIEW

File: crypto-pgpainless/src/main/kotlin/dev/msfjarvis/aps/crypto/PGPKeyManager.kt [hunk 1]
```diff
@@ -50,63 +63,105 @@ public class PGPKeyManager(
...
       }
     }
 
-  override suspend fun getAllKeys(): Result<List<PGPKeyPair>, Throwable> =
+  override suspend fun getAllKeys(): Result<List<Key>, Throwable> =
     withContext(dispatcher) {
       runCatching {
         if (!keyDirExists()) throw KeyManagerException.KeyDirectoryUnavailableException
         val keys = keyDir.listFiles()
         if (keys.isNullOrEmpty()) return@runCatching listOf()
```
Comment (SphericalKat) [inline crypto-pgpainless/src/main/kotlin/dev/msfjarvis/aps/crypto/PGPKeyManager.kt]: could be renamed to `keyFiles` to maintain consistency with `getKeyById`

File: crypto-pgpainless/src/main/kotlin/dev/msfjarvis/aps/crypto/PGPKeyManager.kt [hunk 2]
```diff
@@ -50,63 +63,105 @@ public class PGPKeyManager(
...
+  private fun tryParseKeyring(key: Key): PGPKeyRing? {
+    val secKeyRing = runCatching { PGPainless.readKeyRing().secretKeyRing(key.contents) }.get()
+    if (secKeyRing != null) {
+      return secKeyRing
+    }
+    val pubKeyRing = runCatching { PGPainless.readKeyRing().publicKeyRing(key.contents) }.get()
+    if (pubKeyRing != null) {
+      return pubKeyRing
+    }
+    return null
```
Comment (SphericalKat) [inline crypto-pgpainless/src/main/kotlin/dev/msfjarvis/aps/crypto/PGPKeyManager.kt:134]: Could be simplified to `return pubKeyRing`

File: crypto-pgpainless/src/main/kotlin/dev/msfjarvis/aps/crypto/PGPKeyManager.kt (hunk 1 above)
Comment (msfjarvis) [reply crypto-pgpainless/src/main/kotlin/dev/msfjarvis/aps/crypto/PGPKeyManager.kt]: Done

File: crypto-pgpainless/src/main/kotlin/dev/msfjarvis/aps/crypto/PGPKeyManager.kt (hunk 2 above)
Comment (msfjarvis) [reply crypto-pgpainless/src/main/kotlin/dev/msfjarvis/aps/crypto/PGPKeyManager.kt:134]: I prefer the consistency with `secKeyRing`

**Review by Skrilltrax — APPROVED** [review]

**Pull request merged**